pallets/automation-time/src/lib.rs

```diff
@@ -949,32 +1059,23 @@ pub mod pallet {
 				return Ok(task_id)
 			}
 
-			let outcome = with_transaction(
-				|| -> storage::TransactionOutcome<Result<T::Hash, DispatchError>> {
-					for time in execution_times.iter() {
-						match Self::get_scheduled_tasks(*time) {
-							None => {
-								let task_ids: BoundedVec<T::Hash, T::MaxTasksPerSlot> =
-									vec![task_id].try_into().unwrap();
-								<ScheduledTasks<T>>::insert(*time, task_ids);
-							},
-							Some(mut task_ids) => {
-								if let Err(_) = task_ids.try_push(task_id) {
-									return Rollback(Err(DispatchError::Other("time slot full")))
-								}
-								<ScheduledTasks<T>>::insert(*time, task_ids);
-							},
+			for time in execution_times.iter() {
```

Comment:
Trying to understand this diff to resolve a merge conflict.

Isn't the `with_transaction` pretty relevant here so that we don't partially schedule a user's task while returning an error?

Reply:
This line makes it a transaction. 

https://github.com/OAK-Foundation/OAK-blockchain/pull/146/files#diff-e6a13f7032a383e19b77d20043ca46041d94f5e49396812e2de1b2000c301e59R399